build/package/docker/repo/amazonlinux2-ent.Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM amazonlinux:2
+
+ARG package
+ARG entrypoint
+ARG server_version
+
+RUN printf "[mongodb-enterprise-${server_version}]\nname=MongoDB Enterprise Repository\nbaseurl=https://repo.mongodb.com/yum/amazon/2/mongodb-enterprise/${server_version}/\$basearch/\ngpgcheck=1\nenabled=1\ngpgkey=https://www.mongodb.org/static/pgp/server-${server_version}.asc\n" > /etc/yum.repos.d/mongodb-enterprise-${server_version}.repo
```

Comment:
[nit] you can make community or enterprise an arg instead of having two images
see https://github.com/docker-library/mongo/blob/666ae6bd40f658bbdcb47587289572374756e031/5.0/Dockerfile#L86-L89